services/gomanager/gomanger.go

```diff
@@ -0,0 +1,40 @@
+package gomanager
+
+import (
+	"sync/atomic"
+)
+
+// GoManager ensure the complete execution of go rountes managed by it.
+type GoManager struct {
+	goroutineCount      int32
+	goroutineExitSignal chan struct{}
+}
+
+func New() *GoManager {
+	return &GoManager{
+		goroutineExitSignal: make(chan struct{}, 1),
```

Comment:
I would have loved to get around a `New` method, but I don't think there is a good way to create a buffer channel by default. Maybe using `sync.Once` is an idea. 0/5 on the performance impact using that.

Reply:
I've went with this idea to make the package easier to use.  @agnivade Please let me know what you think,